preview/MsixCore/MsixCoreInstallerLib/inc/IPackageManager.hpp

```diff
@@ -6,13 +6,14 @@
 #include "IPackage.hpp"
 #include "DeploymentOptions.hpp"
 
-
 namespace MsixCoreLib {
     class IPackageManager
     {
     public:
         virtual std::shared_ptr<IMsixResponse> AddPackageAsync(const std::wstring & packageFilePath, DeploymentOptions options, std::function<void(const IMsixResponse&)> callback = nullptr) = 0;
+        virtual std::shared_ptr<IMsixResponse> AddPackageAsync(IStream * packageStream, DeploymentOptions options, std::function<void(const IMsixResponse&)> callback = nullptr) = 0;
         virtual HRESULT AddPackage(const std::wstring & packageFilePath, DeploymentOptions options) = 0;
+        virtual HRESULT AddPackage(IStream * packageStream, DeploymentOptions options) = 0;

```

Comment:
>IStream [](start = 35, length = 7)

can you give us some context or insight on why you wanted to support IStream on this interface as well. i.e. what scenarios are you thinking about where filepath by itself insufficient?

while I think the changes are fine, one concern is that
win10 PackageManager doesn't expose any interfaces that take an IStream. We want installs of MSIX packages to go through the appxdeploymentserver stack on OS's that support MSIX (RS3+) so taking in a stream here means we'd have to download the package to a file in order to pass onto the RS3+ PackageManager APIs.

Reply:
My main goal is to be able to create a self-extractor installer, with a msix package embedded as a resource, so instead of extracting the resource, save it to %Temp% and call AddPackage, we can directly install the msix from the resource stream.

"We want installs of MSIX packages to go through the appxdeploymentserver stack on OS's that support MSIX (RS3+)"

This is interesting to know! It really depends where you want to add the abstraction layer and how you plan to manage this integration. In MsixInstaller.exe or in MsixInstallerLib.lib? I suppose in the executable, if it's the case, it should not be an issue, otherwise, I agree, it would be problematic.

Reply:
We discussed internally, yes we do want both the exe and the lib to be able to redirect to PackageManager. So it seems like for the embedded resource case we'd have to extract it to a temp file and call AddPackage from within the lib.